src/multiint.cpp

```diff
@@ -2458,6 +2485,116 @@ static void loadMapSettings2()
 	}
 }
 
+static void randomizeLimit(const char *name)
+{
+	int stat = getStructStatFromName(name);
+	if (rand() % 2 == 0)
+	{
+		asStructureStats[stat].upgrade[0].limit = asStructureStats[stat].base.limit;
+	}
+	else
+	{
+		asStructureStats[stat].upgrade[0].limit = 0;
+	}
+}
+
+/* Generate random options */
+static void randomizeOptions()
+{
+	if (ingame.bHostSetup)
+	{
+		bLimiterLoaded = true;
+		if (!locked.scavengers)
+		{
+			game.scavengers = rand() % 2;
+			((MultichoiceWidget *)widgGetFromID(psWScreen, MULTIOP_GAMETYPE))->choose(game.scavengers);
+		}
+		if (!locked.alliances)
+		{
+			game.alliance = rand() % 4;
+			((MultichoiceWidget *)widgGetFromID(psWScreen, MULTIOP_ALLIANCES))->choose(game.alliance);
+		}
+		if (!locked.power)
+		{
+			game.power = rand() % 3;
+			((MultichoiceWidget *)widgGetFromID(psWScreen, MULTIOP_POWER))->choose(game.power);
+		}
+		if (!locked.bases)
+		{
+			game.base = rand() % 3;
+			((MultichoiceWidget *)widgGetFromID(psWScreen, MULTIOP_BASETYPE))->choose(game.base);
+		}
+
+		// Structure limits are simply 0 or max, only for NO_TANK, NO_CYBORG, NO_VTOL, NO_UPLINK, NO_LASSAT.
+		if (!bLimiterLoaded || !asStructureStats)
+		{
+			initLoadingScreen(true);
+			if (resLoad("wrf/limiter_data.wrf", 503))
+			{
```

Comment:
@KJeff01 Resource loading is done here to get `asStructureStats` filled in, it might not be available yet at that point. Is there perhaps a better way?

Reply:
...I see. Never really remember this since I don't set structure limits at all. It's not an issue then.